controllers/dspipeline_params.go

```diff
@@ -630,6 +632,59 @@ func (p *DSPAParams) ExtractParams(ctx context.Context, dsp *dspa.DataSciencePip
 			}
 		}
 
+		if cfg := p.APIServer.CustomKfpLauncherConfig; cfg != nil {
+			if cfg.Name != "" && cfg.Key != "" {
+				if cm, err := util.GetConfigMap(ctx, cfg.Name, p.Namespace, client); err != nil {
+					// If the custom kfp-launcher configmap is not available, that is OK
+					if !apierrs.IsNotFound(err) {
+						log.Error(err, fmt.Sprintf("Encountered error when attempting to fetch ConfigMap: [%s], Error: %v", cfg.Name, err))
+						return err
+					}
+				} else {
+					// Consume all the required information.
+					configMapName := config.CustomKfpLauncherConfigMapName
+					dataValues := util.GetConfigMapValues(cm)
+
+					for _, val := range dataValues {
+						if val != "" {
+							p.DataValue = append(p.DataValue, []byte(val))
+						}
+					}
+
+					customConfigMap := &v1.ConfigMap{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      configMapName,
+							Namespace: p.Namespace,
+							OwnerReferences: []metav1.OwnerReference{
+								{
+									APIVersion:         dsp.APIVersion,
+									Kind:               dsp.Kind,
+									Name:               dsp.Name,
+									UID:                dsp.UID,
+									Controller:         util.BoolPointer(true),
+									BlockOwnerDeletion: util.BoolPointer(true),
+								},
+							},
+						},
+
+						Data: map[string]string{
+							p.APIServer.CustomKfpLauncherConfig.Key: string(bytes.Join(p.DataValue, []byte{})),
+						},
+					}
+
+					err := client.Create(ctx, customConfigMap)
```

Comment:
unless I'm totally missing something, this code is writing out a configmap that's not named kfp-launcher.

I also agree with Humair above -- we should be able to read the user-provided configmap into Params, and then use Params to template out the kfp-launcher configmap via manifestival. The way you are attempting to do it here is by writing the configmap using client-go -- which, normally that's how you'd do something in go code, but it's not really the pattern we're using here.

@amadhusu let's pair up when you're back and see if we can work through this together.